core-rust/state-manager/src/staging/sorted_kv_merge_iterator.rs

```diff
@@ -0,0 +1,116 @@
+use std::cmp::Ordering;
+use std::iter::Peekable;
+
+/// Merges two ordered (by K) iterators of (K, V) pairs.
```

Comment:
Sadly this doesn't work right - we need to also handle _deletes_ - which means that the staging store (and this merger) will need to operate on DbOperations rather than values IMO.

In particular, eg, if overlay is delete and base has value, then both iterators advance and ignore the value.